src/pages/Card/_components/SidebarFilter.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+
+import CheckBox from '@/pages/Card/_components/CheckBox';
+import DropDownBox from '@/pages/Card/_components/DropDownBox';
+import { CHECK_BOX_DATA } from '@/pages/Card/_constants/cardData';
+
+import CardFinder from '@/assets/svg/img_cardfinder.svg?react';
+
+function SidebarFilter() {
+  const [checkedState, setCheckedState] = useState(false);
+  const [activeCategories, setActiveCategories] = useState(
+    Array(CHECK_BOX_DATA.length).fill(false),
+  );
+  const [checkboxStates, setCheckboxStates] = useState(
+    CHECK_BOX_DATA.map((category) => Array(category.checkboxes.length).fill(false)),
+  );
+
+  // 전체보기 체크박스 상태 변경 함수
+  const handleCheckChange = () => {
+    setCheckedState(!checkedState);
+  };
```

Comment:
이전 상태를 기반으로 상태를 업데이트 할 때는 setter 함수의 prev 상태를 받아와서 업데이트를 해줘야 예상된 동작대로 상태가 업데이트가 된답니다!
prev 상태는 항상 최신 상태임을 보장하지만 `setCheckedState(!checkedState);` 방식은 스케줄링에 따라서 최신상태 보장이 안될 수 도 있어서 이 부분 수정하시는것도 좋은 방법일것 같아요~~ 다른 부분은 잘 적용이 되어있더라구요 :)

```suggestion
  const handleCheckChange = () => {
    setCheckedState((prev) => !prev );
  };
```

Reply:
오오 저도 이전 상태를 업데이트 하는 함수가 캐러셀이 있는데 이렇게 prev를 통해서 동작하는 방식으로 수정하도록 하겠습니다!

Reply:
앗 넵!! 수정했습니다! 감사합니다!